### ISSUE TITLE
when clause in the case label

### ISSUE THREAD

**Issue by lorcQc** [paul1956/CSharpToVB#49]
### Input
```
Object o = "demo";

switch (o)
{
    case IEnumerable<string> s :
        Console.WriteLine("1"); break;
    case string s when s.Length > 5 :
        Console.WriteLine("2"); break;
    case string s :
        Console.WriteLine("3"); break;		
    default:
        Console.WriteLine("4"); break;
}
//result : 3
```
### Actual output
```
Private o As [Object] = "demo"

Select Case True
    Case TypeOf o Is IEnumerable(Of String)
        Dim s As IEnumerable(Of String) = CType(o, IEnumerable(Of String))
        Console.WriteLine("1")
    Case TypeOf o Is String
        Dim s As String = CType(o, String)
        If Not TypeOf o Is String Then Exit Select
        Console.WriteLine("2")
    Case TypeOf o Is String
        Dim s As String = CType(o, String)
        Console.WriteLine("3")
    Case Else
        Console.WriteLine("4")
End Select
'!!! result : 2
```
### Proposal

add a "TODO Check : " comment to the generated code

**Comment by paul1956** [comment]
@lorcQcI hope to publish a major upgrade this week to the converter with 100's of new tests, I have tests (currently disabled) to check When Clause, today most fail.  I think I fixed this issue but if not I will add TODO.

**Comment by paul1956** [comment]
Fixed in 4.6.0.0/4.0.0.0, to be release by 9/4/2020.
